src/titiler/xarray/titiler/xarray/dependencies.py

```diff
@@ -0,0 +1,211 @@
+"""titiler.xarray dependencies."""
+
+from dataclasses import dataclass
+from typing import Optional, Union
+
+import numpy
+from fastapi import Query
+from rio_tiler.types import RIOResampling, WarpResampling
+from starlette.requests import Request
+from typing_extensions import Annotated
+
+from titiler.core.dependencies import DefaultDependency
+
+
+@dataclass
+class XarrayIOParams(DefaultDependency):
+    """Dataset IO Options."""
+
+    group: Annotated[
+        Optional[int],
+        Query(
+            description="Select a specific zarr group from a zarr hierarchy. Could be associated with a zoom level or dataset."
+        ),
+    ] = None
+
+    reference: Annotated[
+        Optional[bool],
+        Query(
+            title="reference",
+            description="Whether the dataset is a kerchunk reference",
+        ),
+    ] = None
+
+    decode_times: Annotated[
+        Optional[bool],
+        Query(
+            title="decode_times",
+            description="Whether to decode times",
+        ),
+    ] = None
+
+    consolidated: Annotated[
+        Optional[bool],
+        Query(
+            title="consolidated",
+            description="Whether to expect and open zarr store with consolidated metadata",
+        ),
+    ] = None
+
+    # cache_client
+
+
+@dataclass
+class XarrayDsParams(DefaultDependency):
+    """Xarray Dataset Options."""
+
+    variable: Annotated[str, Query(description="Xarray Variable name")]
+
+    drop_dim: Annotated[
+        Optional[str],
+        Query(description="Dimension to drop"),
+    ] = None
+
+    datetime: Annotated[
+        Optional[str], Query(description="Slice of time to read (if available)")
+    ] = None
+
+
+@dataclass
+class XarrayParams(XarrayIOParams, XarrayDsParams):
+    """Xarray Reader dependency."""
+
+    pass
+
+
+@dataclass(init=False)
+class CompatXarrayParams(DefaultDependency):
```

Comment:
this is not directly used in titiler.xarray but could be in a Tiler that would want to support both GDAL/Xarray dataset